include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,277 @@ class CD3DX12_NODE_MASK_SUBOBJECT
     D3D12_NODE_MASK m_Desc;
 };
 
+//------------------------------------------------------------------------------------------------
+// Peter: Implementation for the new CheckFeatureSupport API
+
+#define FEATURE_SUPPORT_GET(RETTYPE,FEATURE,OPTION) \
+RETTYPE OPTION() const \
+{ \
+    return FEATURE.OPTION; \
+}
+
+#define FEATURE_SUPPORT_GET_NAME(RETTYPE,FEATURE,OPTION,NAME) \
+RETTYPE NAME() const \
+{\
+    return FEATURE.OPTION; \
+}
+
+#define FEATURE_SUPPORT_GET_NODE_INDEXED(RETTYPE,FEATURE,OPTION) \

```

Comment:
Macro to set up getter functions for features indexed by NodeIndex. Default to 0, or the first available graphics device node.